bindings/java/src/main/java/org/apache/opendal/Operator.java

```diff
@@ -50,17 +49,24 @@ public Operator(String schema, Map<String, String> params) {
 
     private native void write(long ptr, String fileName, String content);
 
+    private native void asyncWrite(long ptr, String fileName, String content, CompletableFuture<Boolean> future);
+
     private native String read(long ptr, String fileName);
 
     private native void delete(long ptr, String fileName);
 
     private native long stat(long ptr, String file);
 
-
     public void write(String fileName, String content) {
         write(this.ptr, fileName, content);
     }
 
+    public CompletableFuture<Boolean> asyncWrite(String fileName, String content) {
```

Comment:
How will java community design API for both blocking and async operations? Do they prefer to have `asyncWrite` to a seperate class called `AsyncOperator`?

Reply:
I think both of them are ok. [kafka consumer](https://github.com/apache/kafka/blob/d83a734c41c43a03aa571e602cf8535f0893fd79/clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java#L105-L112)  provides blocking and async function in one class. But aws s3 client spit them into [DelegatingS3Client](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/DelegatingS3Client.html) and [DelegatingS3AsyncClient](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/DelegatingS3AsyncClient.html).

Reply:
We can provide sync and async functions in the OperatorImpl class and make it implement SyncOperator and AsyncOperator interfaces. So that users could make decisions by themselves.

Reply:
> We can provide sync and async functions in the OperatorImpl class and make it implement SyncOperator and AsyncOperator interfaces.

I prefer the simpler solution. `writeSync` and `writeAsync` LGTM.